### PR TITLE
use 'auto' instead of 1 for spectral rechunk

### PR DESCRIPTION
**Pull request by keflavich**
Minor correction to partly address #729

### CODE REVIEW

**Comment by keflavich** [comment]
this apparently breaks!

**Comment by keflavich** [comment]
This shows that apparently `spectral_interpolate` is changing the cube size in the _spatial_ dimensions.  ?!?!?
image

**Comment by keflavich** [comment]
I switched from `np.interp` to `scipy.interpolate.interp1d` because the latter allows for straightforward broadcasting, while the former does not - in our non-dask version, we have a loop over valid spectral pixels.  This seems to work.

File: spectral_cube/dask_spectral_cube.py [hunk 1]
```diff
@@ -1265,21 +1266,26 @@ def spectral_interpolate(self, spectral_grid,
...
             else:
-                return np.interp(args[0], args[1], y[:, 0, 0],
-                                 left=fill_value, right=fill_value).reshape((-1, 1, 1))
-
-        if reverse_in:
-            cubedata = cubedata[::-1, :, :]
-
-        cubedata = cubedata.rechunk((-1, 1, 1))
+                #return np.interp(args[0], args[1], y[:, 0, 0],
+                #                 left=fill_value, right=fill_value).reshape(chunkshape)
```
Comment (e-koch) [inline spectral_cube/dask_spectral_cube.py:1280]: Clean up here.

Comment (e-koch) [reply spectral_cube/dask_spectral_cube.py:1280]:
```suggestion

```

**Review by e-koch — APPROVED** [review]
Looks good, just a small clean-up.

I don't think the tests need to change from the existing ones, since they serve the same purpose.

The dask and non-dask versions (so scipy vs numpy now) are equivalent from a couple local tests I ran.

Note: In a separate PR, we should make scipy a dependency. The non-dask spectral interpolate could be switched over just to match here, but there's currently no functional reason to do so.

**Pull request merged**